ssl/quic/quic_rx_frame.c

```diff
@@ -0,0 +1,391 @@
+        case OSSL_QUIC_FRAME_TYPE_ACK_WITH_ECN:
+        case OSSL_QUIC_FRAME_TYPE_CONN_CLOSE_TRANSPORT:
+        case OSSL_QUIC_FRAME_TYPE_CONN_CLOSE_APP:
+            break;
+        default:
+            /* This frame made the packet ACK eliciting */
+            ackm_data->is_ack_eliciting = 0;
+            break;
+        }
+    }
+}
+
+static int depack_process_frames(PACKET *pkt, int packet_space,
+                                 OSSL_TIME received)
+{
+    while (PACKET_remaining(&pkt) > 0) {
+        if (!ossl_quic_wire_peek_frame_header(&pkt, &frame_type))
+            return 0;
+
+        switch (frame_type) {
+        case OSSL_QUIC_FRAME_TYPE_PADDING:
+        case OSSL_QUIC_FRAME_TYPE_PING:
+            /*
+             * We ignore this.
+             * For the PING frame, eliciting an ACK is enough.
+             * Since the PING frame is only a header, we simply
+             * skip it.
+             */
+            ossl_quic_wire_skip_frame_header(&pkt, &frame_type);
+            break;
+
+        case OSSL_QUIC_FRAME_TYPE_ACK_WITHOUT_ECN:
+        case OSSL_QUIC_FRAME_TYPE_ACK_WITH_ECN: {
+            OSSL_ACKM_ACK_RANGE ranges[20];
+            size_t total_ranges;
+            OSSL_ACKM_ACK ack;
+
+            ack.ack_ranges = ranges;
+            ack.num_ack_ranges = OSSL_NELEM(ranges);
+            /* QUESTION: where do I get the ack delay exponent? */
+            if (!ossl_quic_wire_decode_frame_ack(pkt, ack_delay_exponent,
+                                                 &ack, NULL))
+                return 0;
+            if (!ossl_ackm_on_rx_ack_frame(connection->ackm, &ack,
+                                           packet_space, received))
+                return 0;
+            break;
+        }
+
+        case OSSL_QUIC_FRAME_TYPE_RESET_STREAM: {
+            OSSL_QUIC_FRAME_RESET_STREAM frame_data;
+            SSL *stream;
+            int stream_type;
+
+            if (!ossl_quic_wire_decode_frame_reset_stream(pkt, &frame_data))
+                return 0;
+
+            /*
+             * ASSUMPTION: ssl_get_stream() gets a stream from a connection
+             * by stream ID.
+             */
+            stream = ssl_get_stream(connection, frame_data.stream_id);
+            stream_type = SSL_get_stream_type(stream);
+
+            /*
+             * ASSUMPTION: ssl_close_stream() detaches the stream SSL from
+             * the connection SSL it's attached to, and then destroys that
+             * stream SSL.
+             */
+            ssl_close_stream(stream);
+
+            if (stream_type == SSL_STREAM_TYPE_WRITE) {
+                /*
+                 * ASSUMPTION: ssl_close_connection() closes all the streams
+                 * that are attached to it, then destroys the connection SSL.
+                 */
+                ssl_close_connection(packet->connection);
+                packet->connection = NULL;
+                ossl_quic_set_state(???, QUIC_STREAM_STATE_ERROR);
```

Comment:
Probably we need something like `SSLfatal` to close a connection in error and raise appropriate errors.

Reply:
Yes.  I've added a temporary static function for now, to be determined what it should be replaced with